tests/Unit/LocatorTest.php

```diff
@@ -136,13 +136,16 @@ public function test_from_file(File $data): void
 
         $expected = [];
         $document = new DOMDocument();
-        $document->loadHTML($data->body);
+        $document->loadHTML((string) $data->body);
         $xpath = new DOMXPath($document);
-        foreach ($xpath->query('//link') as $element) {
-            /** @var \DOMElement $element */
-            $expected[] = 'http://example.com' . $element->getAttribute('href');
+
+        if ($queryResult = $xpath->query('//link')) {
```

Comment:
I think it is clearer to use `@var` for the cases where PHPStan analysis is deficient, rather than to add conditionals that will always succeed.

So until https://github.com/phpstan/phpstan/issues/9553 is implemented, I would do something like:

```php
/** @var \DOMNodeList $links */
$links = $xpath->query('//link');
foreach ($links as $element) {
```

We might even do `DOMNodeList<DOMElement>` and then the second `@var` should not be necessary.

Reply:
Got it, done.